datafusion-cli/src/object_storage.rs

```diff
@@ -188,6 +217,8 @@ pub struct AwsOptions {
     pub region: Option<String>,
     /// OSS or COS Endpoint
     pub endpoint: Option<String>,
+    /// Allow HTTP (otherwise will always use https)
```

Comment:
BTW I found the reworked configuration option system that @metesynnada  added in https://github.com/apache/arrow-datafusion/pull/9382 really nice to work with ❤️